### ISSUE TITLE
Value of nodeID.max()=-1 in encoderTool.py throws an error

### ISSUE THREAD

**Issue by anustupc** [zb12138/OctAttention#18]
Hi, 
When trying to encode my own point cloud, I found that I am getting an error in the line 
`bit0,acc0 =encodeNode(proBit[i],octvalue)`
in the compress function of the encoderTool.py file. 

This is happening because the proBit list is empty. While digging deeper, I found that this is occurring because nodeID.max() is -1 in 
`proBit.append(pro[:nodeID.max()+1].detach().cpu().numpy())`

Can you kindly tell why that might be happening and if there is some way around that?

Thanks

**Comment by zb12138** [comment]
Hello I noticed this problem before. This problem may occur with point clouds with some particular numer of points. But I'm too busy recently, can you share your data and help me fix the bug? Thank you.

**Comment by anustupc** [comment]
Hi, 
The data is internal and unfortunately, I won't be able to share it. But here is some change that I did to make it work (for now) - To not make the encoder crash, I changed the lines from 
```
if( (n % MAX_GPU_MEM_It==0 and n>0) or n == padingLength//bptt-1):
               proBit.append(pro[:nodeID.max()+1].detach().cpu().numpy())
               offset = offset + nodeID.max() +1
```
to 
```
if (nodeID.max().detach().cpu().numpy())>-1:
                temp = pro[:nodeID.max()+1]
if( (n % MAX_GPU_MEM_It==0 and n>0) or n == padingLength//bptt-1):
    proBit.append(temp.detach().cpu().numpy())
    offset = offset + nodeID.max() +1
```
 
Not sure, if it makes sense but I have yet to test the decoder part
